packages/bot-skeleton/src/services/api/ticks_service.js

```diff
@@ -217,16 +224,20 @@ export default class TicksService {
     requestTicks(options) {
         const { symbol, granularity, style } = options;
 
-        return new Promise((resolve, reject) => {
-            doUntilDone(() =>
+        if (!this.tick_history_promise) {
```

Comment:
Same here, instead of creating a new `Promise`, just return the one we have in memory (granted we have one).

Reply:
done